drivers/wifi/nrf700x/src/wifi_mgmt.c

```diff
@@ -948,3 +948,67 @@ int nrf_wifi_filter(const struct device *dev,
 	return ret;
 }
 #endif /* CONFIG_NRF700X_RAW_DATA_RX || CONFIG_NRF700X_PROMISC_DATA_RX */
+
+int nrf_wifi_set_rts_threshold(const struct device *dev,
+			       unsigned int rts_threshold)
+{
+	enum nrf_wifi_status status = NRF_WIFI_STATUS_FAIL;
+	struct nrf_wifi_ctx_zep *rpu_ctx_zep = NULL;
+	struct nrf_wifi_vif_ctx_zep *vif_ctx_zep = NULL;
+	struct nrf_wifi_umac_set_wiphy_info wiphy_info;
+	int ret = -1;
+
+	if (!dev) {
+		LOG_ERR("%s: dev is NULL", __func__);
+		return ret;
+	}
+
+	vif_ctx_zep = dev->data;
+
+	if (!vif_ctx_zep) {
+		LOG_ERR("%s: vif_ctx_zep is NULL", __func__);
+		return ret;
+	}
+
+	rpu_ctx_zep = vif_ctx_zep->rpu_ctx_zep;
+
+	if (!rpu_ctx_zep) {
+		LOG_ERR("%s: rpu_ctx_zep is NULL", __func__);
+		return ret;
+	}
+
+	k_mutex_lock(&vif_ctx_zep->vif_lock, K_FOREVER);
+
+	if (!rpu_ctx_zep->rpu_ctx) {
+		LOG_DBG("%s: RPU context not initialized", __func__);
+		goto out;
+	}
+
+	memset(&wiphy_info, 0, sizeof(struct nrf_wifi_umac_set_wiphy_info));
+
+	wiphy_info.rts_threshold = rts_threshold;
+
+	if (wiphy_info.rts_threshold < -1) {
```

Comment:
is this not throwing any warning as the casts are missing?

Reply:
Updated, somehow it was not showing any warning.